resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -180,10 +184,14 @@ private[spark] class KubernetesSparkRestServer(
                 resolvedSparkProperties("spark.jars") = resolvedJars.mkString(",")
 
                 // Resolve spark.files
-                val originalFiles = sparkProperties.get("spark.files")
+                val nonUploadedFiles = sparkProperties.get("spark.files")
                   .map(_.split(","))
+                  .map(_.filter(Utils.resolveURI(_).getScheme match {
```

Comment:
Using the above we can shorthand a lot of things and avoid the case-match entirely. For example:

`Option(Utils.resolveURI(file).getScheme).getOrElse("file") != "file")`